DirectX11Proj/IApplication.cpp

```diff
@@ -1,34 +1,62 @@
 #include "IApplication.h"
 
+#include <iostream>
+
+
 #include "IScene.h"
 #include "systemclass.h"
 
-IApplication::IApplication() : currentScene(nullptr), shouldQuit(false)
+
+IApplication::IApplication() : mpCurrentScene(nullptr), mShouldQuit(false)
 {
 }
 
 
 IApplication::~IApplication()
 {
+	if (mpCurrentScene)
+	{
+		// fucking evil (wow). should really fix something about this (and think of a decent scene management system)
+		delete mpCurrentScene;
+	}
 }
 
-
-void IApplication::SetCurrentScene(IScene* aScene)
+// Setting the current scene also initializes it
+void IApplication::LoadScene(IScene* aScene)
 {
+	assert(aScene != nullptr);
+
+	if (aScene == nullptr)
+	{
+		std::cout << "Can not set a nullptr as current scene!" << std::endl;
+		return;
+	}
 
+	if (mpCurrentScene != nullptr)
+	{
+		std::cout << "There is already a scene loaded!" << std::endl;
```

Comment:
fair enough, its a bit wasteful but a decent thing to do for my one scene system